spyder/utils/code_analysis/client.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+
+"""Spyder MS Language Server v3.0 client implementation."""
+
+import os
+import sys
+import zmq
+import json
+import time
+import socket
+import logging
+import argparse
+import subprocess
+import coloredlogs
+import os.path as osp
+from threading import Thread, Lock
+from spyder.py3compat import PY2, getcwd
+from spyder.utils.code_analysis import EDITOR_CAPABILITES, TRACE
+
+if PY2:
+    import pathlib2 as pathlib
+else:
+    import pathlib
+
+
+TIMEOUT = 5000
+PID = os.getpid()
+
+
+parser = argparse.ArgumentParser(
+    description='ZMQ Python-based MS Language-Server v3.0 client for Spyder')
```

Comment:
I have no opinion, if using zmq makes the core easier to read and write, then why not?